arbnode/batch_poster.go

```diff
@@ -297,19 +304,20 @@ func (s *batchSegments) addSegmentToCompressed(segment []byte) error {
 }
 
 // returns false if segment was too large, error in case of real error
-func (s *batchSegments) addSegment(segment []byte, isHeader bool) (bool, error) {
+func (s *batchSegments) addSegment(segment []byte, isHeader bool, forceInclude bool) (bool, error) {
 	if s.isDone {
 		return false, errBatchAlreadyClosed
 	}
 	err := s.addSegmentToCompressed(segment)
 	if err != nil {
 		return false, err
 	}
```

Comment:
you can tell if this is the first useful message here without requiring a parameter by comparing len(s.rawSegments) with s.trailingHeaders.. just an option, not a must.

Reply:
Done, thanks for the recommendation!